### PR TITLE
[feature] Observer pattern implemented in notifications

### PR DESCRIPTION
**Pull request by jccolina**
Description
===========
- Device service logic implemented
- Handling of notifications implemented in school service
@JurgenPadilla

### CODE REVIEW

File: src/colinajose/service/SchoolService.java [hunk 1]
```diff
@@ -46,12 +49,16 @@ public String registerCourse(String grade, String year, String group){
     public String registerParent(String name, String ci, int age, String gender, String address, String phone, String email){
         Contact contact = new Contact(address, phone, email);
         Parent parent = new Parent(name, ci, age, gender, contact);
-        school.addParent(parent);
+        Device device =  new Device(parent);
```
Comment (JurgenPadilla) [inline src/colinajose/service/SchoolService.java]: Is not mandatory for the parents to have a Device. So registerDevice should be another action available.

File: src/colinajose/service/SchoolService.java [hunk 2]
```diff
@@ -61,20 +68,73 @@ public String registerTeacher(String name, String ci, int age, String gender, St
...
+        students.addAll(getNotifyStudents(kardexId));
+        return getDevices(students);
+    }
+    private MyCircularDoublyLinkedList<Device> getDevices(MyLinkedList<Student> students){
+        MyCircularDoublyLinkedList<Device> devices = new MyCircularDoublyLinkedList<>();
+        for (int i = 0; i < students.size(); i++) {
+            Student student = students.get(i);
+            MyCircularDoublyLinkedList<Person> parents = student.getParents();
+            for (int j = 0; j < parents.size(); j++) {
+                Device device = SearchService.getDevice(this.school, parents.get(i));
```
Comment (JurgenPadilla) [inline src/colinajose/service/SchoolService.java]: Not sure I like this. One person can have many devices. Meaning that the owner can reference to the same parent for multiple devices.
This should be getDevices and return a Collection of devices.

**Review by JurgenPadilla — CHANGES_REQUESTED** [review]

File: src/colinajose/service/SchoolService.java (hunk 1 above)
Comment (jccolina) [reply src/colinajose/service/SchoolService.java]: @JurgenPadilla fixed in next commit

File: src/colinajose/service/SchoolService.java (hunk 2 above)
Comment (jccolina) [reply src/colinajose/service/SchoolService.java]: @JurgenPadilla fixed in next commit

**Review by JurgenPadilla — APPROVED** [review]

**Pull request merged**